src/shaka_scheme/system/base/Vector.hpp

```diff
@@ -0,0 +1,81 @@
+#ifndef SHAKA_SCHEME_CORE_BASE_VECTOR_HPP
+#define SHAKA_SCHEME_CORE_BASE_VECTOR_HPP
+
+#include "shaka_scheme/system/base/Data.hpp"
+
+#include <initializer_list>
+#include <cctype>
+
+
+namespace shaka {
+
+class Vector {
+  NodePtr* arr;

```

Comment:
Is there a specific reason that you preferred a raw pointer to a dynamically allocated array of NodePtr instead of an `std::vector<NodePtr>`? I'm mostly just curious.

Reply:
Yes -- `shaka::Vector` is always going to be a fixed size -- it cannot be appended to and so any functions that would "make it bigger" would simply return a bigger `shaka::Vector` and therefore reallocate anyways.